pandas/core/internals.py

```diff
@@ -2451,6 +2462,182 @@ def shift(self, periods, axis=0, mgr=None):
                                            placement=self.mgr_locs)]
 
 
+class PeriodBlock(NonConsolidatableMixIn, DatetimeBlock):
```

Comment:
you might be better actoring out some of the holding Index suff from ``DatetimeTZBLock`` and creating ``IndexHolderBlock`` (or better name). then sub-classing for ``DatetimeTZBlock`` and ``PeriodBlock``. Should be in theory less duplication.